src/GraphQL.Harness/Startup.cs

```diff
@@ -24,24 +25,36 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            // add execution components
             services.AddSingleton<IDocumentExecuter, DocumentExecuter>();
             services.AddSingleton<IDocumentWriter, DocumentWriter>();
 
+            // add something like repository
             services.AddSingleton<StarWarsData>();
+
+            // add graph types
             services.AddSingleton<StarWarsQuery>();
             services.AddSingleton<StarWarsMutation>();
             services.AddSingleton<HumanType>();
             services.AddSingleton<HumanInputType>();
             services.AddSingleton<DroidType>();
             services.AddSingleton<CharacterInterface>();
             services.AddSingleton<EpisodeEnum>();
+
+            // add schema
             services.AddSingleton<ISchema, StarWarsSchema>();
 
+            // add infrastructure stuff
             services.AddHttpContextAccessor();
             services.AddLogging(builder => builder.AddConsole());
 
+            // add options configuration
             services.Configure<GraphQLSettings>(Configuration);
             services.Configure<GraphQLSettings>(settings => settings.BuildUserContext = ctx => new GraphQLUserContext { User = ctx.User });
+
+            // add Field Middlewares
+            services.AddSingleton<CountFieldMiddleware>();
+            services.AddSingleton<InstrumentFieldsMiddleware>();
```

Comment:
Just a note - this line of code should be done in the server project.

Reply:
@Shane32 Strange, but without this registration in the server project, the middleware still works. I don’t understand why this is happening.

Reply:
I figured it out - `DefaultServiceProvider`